src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingFailureSpec.cs

```diff
@@ -180,35 +181,51 @@ private void StartSharding()
                 shardResolver: extractShardId);
         }
 
-        [MultiNodeFact(Skip = "TODO")]
-        public void ClusterSharding_with_flaky_journal_should_setup_shared_journal()
+        [MultiNodeFact]
+        public void ClusterSharding_with_flaky_journal_network_specs()
+        {
+            ClusterSharding_with_flaky_journal_network_should_setup_shared_journal();
+            ClusterSharding_with_flaky_journal_network_should_join_cluster();
+            ClusterSharding_with_flaky_journal_network_should_recover_after_journal_network_failure();
+        }
+
+        public void ClusterSharding_with_flaky_journal_network_should_setup_shared_journal()
         {
             // start the Persistence extension
             Persistence.Persistence.Instance.Apply(Sys);
             RunOn(() =>
             {
-                Sys.ActorOf(Props.Create<MemoryJournal>(), "store");
-            }, _controller);
+                Persistence.Persistence.Instance.Apply(Sys).JournalFor("akka.persistence.journal.MemoryJournal");
+            }, _config.Controller);
             EnterBarrier("persistence-started");
 
             RunOn(() =>
             {
-                Sys.ActorSelection(Node(_first) / "user" / "store").Tell(new Identify(null));
-                var sharedStore = ExpectMsg<ActorIdentity>().Subject;
-                //TODO: SharedLeveldbJournal.setStore(sharedStore, system)
-            }, _first, _second);
+                Sys.ActorSelection(Node(_config.Controller) / "system" / "akka.persistence.journal.MemoryJournal").Tell(new Identify(null));
+                var sharedStore = ExpectMsg<ActorIdentity>(TimeSpan.FromSeconds(10)).Subject;
+                sharedStore.Should().NotBeNull();
+
+                MemoryJournalShared.SetStore(sharedStore, Sys);
+            }, _config.First, _config.Second);
             EnterBarrier("after-1");
+
+            RunOn(() =>
+            {
+                //check persistence running
+                var probe = CreateTestProbe();
+                var journal = Persistence.Persistence.Instance.Get(Sys).JournalFor(null);
+                journal.Tell(new Persistence.ReplayMessages(0, 0, long.MaxValue, Guid.NewGuid().ToString(), probe.Ref));
+                probe.ExpectMsg<Persistence.RecoverySuccess>(TimeSpan.FromSeconds(10));
+            }, _config.First, _config.Second);
+            EnterBarrier("after-1-test");
         }
 
-        [MultiNodeFact(Skip = "TODO")]
-        public void ClusterSharding_with_flaky_journal_should_join_cluster()
+        public void ClusterSharding_with_flaky_journal_network_should_join_cluster()
         {
-            ClusterSharding_with_flaky_journal_should_setup_shared_journal();
-
-            Within(TimeSpan.FromSeconds(20), () =>
+            Within(TimeSpan.FromSeconds(30), () =>
```

Comment:
Why 30 sec, not 20? If 20 sec does not work, please leave a comment about it in the code